src/parser.rs

```diff
@@ -0,0 +1,188 @@
+use arithmetic_problems_generator::{CheckerConfig, Config, GeneratorConfig};
+use clap::ArgMatches;
+
+pub fn to_config(matches: ArgMatches) -> Result<Config, Box<dyn std::error::Error>> {
+    if matches.contains_id("range") {
+        //check the value of range is valid
+        let range = match matches.get_one::<String>("range").unwrap().parse::<u32>() {
+            Ok(value) => value,
+            Err(_) => return Err(format!("argument conversion exception").into()),
+        };
+
+        if matches.contains_id("num") {
+            let num = match matches.get_one::<String>("num").unwrap().parse::<u32>() {
+                Ok(value) => value,
+                Err(_) => return Err(format!("argument conversion exception").into()),
+            };
+            return Ok(Config::Generator(GeneratorConfig::new(num, (0, range))));
+        } else {
+            return Ok(Config::Generator(GeneratorConfig::new(10, (0, range))));
+        }
+    } else if matches.contains_id("num") && !matches.contains_id("range") {
+        let num = match matches.get_one::<String>("num").unwrap().parse::<u32>() {
+            Ok(value) => value,
+            Err(_) => return Err(format!("argument conversion exception").into()),
+        };
+        return Ok(Config::Generator(GeneratorConfig::new(num, (0, 100))));
+    } else if matches.contains_id("exercise_file") && matches.contains_id("answer_file") {
+        let exercise_file = matches.get_one::<String>("exercise_file").unwrap().clone();
+        let answer_file = matches.get_one::<String>("answer_file").unwrap().clone();
+        return Ok(Config::Checker(CheckerConfig::from_str(
+            &exercise_file,
+            &answer_file,
+        )));
+    } else {
+        Err(format!("argument missing").into())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use super::to_config;
+    use arithmetic_problems_generator::{CheckerConfig, Config};
+    use clap::{Arg, ArgGroup, Command};
```

Comment:
Re-order uses. Use `super::*` is a more conventional way.

```suggestion
    use super::*;
    use arithmetic_problems_generator::{CheckerConfig, Config};
    use clap::{Arg, ArgGroup, Command};
    use std::path::PathBuf;
```

Reply:
We don't need to import the all models in the super. If we choose to use `use super::*`, will it lead to import unnecessary models?

Reply:
不会，正式因为引入所有模组，迫使你要对每一个 Public API 进行测试

Reply:
你使用`use super::*` 并不会提示说你没有进行对全部Public API进行测试，我认为你使用`use super::to_config` 更能清楚的看到已经对哪个API进行了测试，尤其是在增加了API的时候 ，正如Java在编写mysql语句的时候 推荐使用的`Select 字段名`而不是`select *`